src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp

```diff
@@ -0,0 +1,484 @@
+/*
+ ISC License
+
+ Copyright (c) 2022, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include "spinningBodyTwoDOFStateEffector.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/rigidBodyKinematics.h"
+#include <string>
+
+/*! This is the constructor, setting variables to default values */
+SpinningBodyTwoDOFStateEffector::SpinningBodyTwoDOFStateEffector()
+{
+    // Zero the mass props and mass prop rates contributions
+    this->effProps.mEff = 0.0;
+    this->effProps.rEff_CB_B.fill(0.0);
+    this->effProps.IEffPntB_B.fill(0.0);
+    this->effProps.rEffPrime_CB_B.fill(0.0);
+    this->effProps.IEffPrimePntB_B.fill(0.0);
+
+    // Initialize variables to working values
+    this->IS1PntSc1_S1.setIdentity();
+    this->IS2PntSc2_S2.setIdentity();
+    this->dcm_S10B.setIdentity();
+    this->dcm_S20S1.setIdentity();
+    this->r_S1B_B.setZero();
+    this->r_S2S1_S1.setZero();
+    this->s1Hat_S1.setZero();
+    this->s2Hat_S2.setZero();
+
+    // Create the output vectors
+    this->spinningBodyConfigLogOutMsgs.push_back(new Message<SCStatesMsgPayload>);
```

Comment:
:small_red_triangle_down: We can just push back the Messages. When the destructor of the vector is called it will call the destructor of the elements.

That being said, we need to move to C++14 (simply a project compiler flag change), and we need to move away from the raw pointer usage throughout the code base. This a small but powerful change, which will give us both `uniqe_ptr` and `shared_ptr`, whose memory is always managed, even when not within in a C++ container type.

Reply:
Done. I was thinking about C-style arrays of pointers. Thanks for the feedback.